src/bots/liquidator.ts

```diff
@@ -600,6 +641,187 @@ export class LiquidatorBot implements Bot {
 			});
 	}
 
+	private getOrderParamsForPerpDerisk(
+		subaccountId: number,
+		position: PerpPosition
+	): OptionalOrderParams | undefined {
+		const nowSec = Math.floor(Date.now() / 1000);
+
+		let baseAssetAmount: BN;
+		if (this.isTwapping()) {
+			const twapProgress = this.twapExecutionProgresses.get(
```

Comment:
what happens if the perp market goes live after the bot starts, wont there not be a twap progress object set?